lib/elixir/lib/calendar.ex

```diff
@@ -417,12 +489,273 @@ defmodule Date do
       2
       iex> Date.day_of_week(~N[2016-11-01 01:23:45])
       2
+
   """
-  @spec day_of_week(Calendar.date) :: non_neg_integer()
+  @spec day_of_week(Calendar.date | Calendar.integer_date) :: non_neg_integer()
   def day_of_week(%{calendar: calendar, year: year, month: month, day: day}) do
     calendar.day_of_week(year, month, day)
   end
 
+  def day_of_week(date) when is_integer(date) do
```

Comment:
I am not sure if we should expose functions such as this. I would rather have developers make the conversion explicit maybe by introducing something such as `Date.from_days(...)` or something similar.